File: 백준/Silver/14501. 퇴사/퇴사.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const n = Number(input[0])
const days = input.slice(1).map((row) => row.split(" ").map(Number))

const dp = Array.from({ length: n }, () => 0)

for (let i = 0; i <n; i++) {
  const [t, p] = days[i]
  if (i + t > n) continue
  dp[i] = dp[i] + p
  for(let j=i+t;j<n;j++){
    dp[j] = Math.max(dp[j],dp[i])
  }
}
console.log(Math.max(...dp))

File: 백준/Silver/14501. 퇴사/퇴사.py
import sys
input = sys.stdin.readline

n = int(input())
counsel = [list(map(int,input().split(' '))) for _ in range(n)]

dp = [0] * n

for i in range(n):
  t,p = counsel[i]
  if i+t>n: continue
  dp[i] = dp[i]+p

  for j in range(i+t,n):
    dp[j] = max(dp[i],dp[j])

print(max(dp))


